xonsh/completers/tools.py

```diff
@@ -31,3 +31,53 @@ def justify(s, max_length, left_pad=0):
     """
     txt = textwrap.wrap(s, width=max_length, subsequent_indent=" " * left_pad)
     return "\n".join(txt)
+
+
+class RichCompletion(str):
```

Comment:
We probably shouldn't subclass string directly. Maybe `collections.UserString` is what you are looking for?

Reply:
Hi!
I'm not sure that would be better, since we *do* want the object to behave like a regular string and be a subclass of it.
For example, at `xonsh/ptk_shell/completer.py:72` the function `os.path.commonprefix` is being called on the completions, which demands them to be `str`s.
Also, according to the [python docs](https://docs.python.org/3/library/collections.html#userstring-objects):

> The need for this class has been partially supplanted by the ability to subclass directly from str; however, this class can be easier to work with because the underlying string is accessible as an attribute

... Which we don't really need

Reply:
Yeah, maybe this is OK now